lib/iris/fileformats/netcdf/loader.py

```diff
@@ -229,21 +237,57 @@ def _get_cf_var_data(cf_var, filename):
             )
             # Get the chunking specified for the variable : this is either a shape, or
             # maybe the string "contiguous".
-            chunks = cf_var.cf_data.chunking()
-            # In the "contiguous" case, pass chunks=None to 'as_lazy_data'.
-            if chunks == "contiguous":
-                chunks = None
-
-            # Return a dask array providing deferred access.
-            result = as_lazy_data(proxy, chunks=chunks)
-
+            if CHUNK_CONTROL.mode is ChunkControl.Modes.AS_DASK:
+                result = as_lazy_data(proxy, chunks=None, dask_chunking=True)
+            else:
+                chunks = cf_var.cf_data.chunking()
+                if (
+                    chunks is None
+                    and CHUNK_CONTROL.mode is ChunkControl.Modes.FROM_FILE
+                ):
+                    raise KeyError(
+                        f"{cf_var.cf_name} does not contain pre-existing chunk specifications."
+                        f"Instead, you might wish to use CHUNK_CONTROL.set(), or just use default"
+                        f" behaviour outside of a context manager. "
+                    )
+                # In the "contiguous" case, pass chunks=None to 'as_lazy_data'.
+                if chunks == "contiguous":
+                    # Equivalent to chunks=None, but value required by chunking control
+                    chunks = list(cf_var.shape)
+
+                # Modify the chunking in the context of an active chunking control.
+                # N.B. settings specific to this named var override global ('*') ones.
+                dim_chunks = CHUNK_CONTROL.var_dim_chunksizes.get(
+                    cf_var.cf_name
+                ) or CHUNK_CONTROL.var_dim_chunksizes.get("*")
+                dims = cf_var.cf_data.dimensions
+                if CHUNK_CONTROL.mode is ChunkControl.Modes.FROM_FILE:
+                    dims_fixed = np.ones(len(dims), dtype=bool)
+                elif not dim_chunks:
+                    dims_fixed = None
+                else:
+                    # Modify the chunks argument, and pass in a list of 'fixed' dims, for
+                    # any of our dims which are controlled.
+                    dims_fixed = np.zeros(len(dims), dtype=bool)
+                    for i_dim, dim_name in enumerate(dims):
+                        dim_chunksize = dim_chunks.get(dim_name)
+                        if dim_chunksize:
+                            if dim_chunksize == -1:
+                                chunks[i_dim] = cf_var.shape[i_dim]
+                            else:
+                                chunks[i_dim] = dim_chunksize
+                            dims_fixed[i_dim] = True
+                if dims_fixed is None:
+                    dims_fixed = [dims_fixed]
```

Comment:
`elif` check for `dims_fixed` being all `False`, which would indicate that user might want chunking but have made a typo/other mistake when using `CHUNK_CONTROL.set()`. A test for the warning as well.

Reply:
Further investigation reveals that this would be triggered whenever a user had used `CHUNK_CONTROL.set()` but the variable being processed was not associated with the dimensions the user was controlling. To nuanced, not an easy convenience to offer.